packages/tabs/src/TabTitle/TabTitle.tsx

```diff
@@ -62,12 +101,15 @@ const TabTitle: ExtendableBox<BaseTabTitleProps, 'button'> = ({
       listTitleModeStyles[theme].focus,
       className,
     ),
+    disabled,
+    id,
+    name: nodeText,
+    onClick: handleClick,
     role: 'tab',
     tabIndex: selected ? 0 : -1,
+    ['aria-controls']: relatedTabPanel?.id,
     ['aria-selected']: selected,
-    name: nodeText,
     ['data-text']: nodeText,
-    disabled,
   } as const;
 
   if (typeof rest.href === 'string') {
```

Comment:
If you're feeling up to it, you could take this opportunity to replace `Box` with `Polymorphic` 😊

Reply:
Or even just consolidating this to 
```jsx
<Box 
  as={typeof rest.href === 'string' ? 'a' : 'button'}
   ...
>
```

Reply:
I've got a separate ticket to handle this so will knock it out with that change! https://jira.mongodb.org/browse/LG-2728